splink/comparison_template_library.py

```diff
@@ -0,0 +1,173 @@
+# Script defining comparison templates that act as wrapper functions which produce
+# comparisons based on the type of data in a column with default values to make
+# it simpler to use splink out-of-the-box
+
+from __future__ import annotations
+
+from .comparison import Comparison  # change to self
+from .comparison_library_utils import datediff_error_logger
+from .misc import ensure_is_iterable
+
+
+class DateComparisonBase(Comparison):
+    def __init__(
+        self,
+        col_name,
+        include_exact_match_level=True,
+        term_frequency_adjustments=False,
+        separate_1st_january=True,
+        levenshtein_thresholds=[2],
+        datediff_thresholds=[[1, 1], ["month", "year"]],
+        m_probability_exact_match=None,
+        m_probability_1st_january=None,
+        m_probability_or_probabilities_lev: float | list = None,
+        m_probability_or_probabilities_datediff: float | list = None,
+        m_probability_else=None,
```

Comment:
A slightly wider point - do we want to allow access to all of the underlying m_probabilities for users? It makes our comparison functions harder to interpret.

Reply:
I thought that, but all the other functions had them so I just replicated them here.

I also wasn't sure exactly why these exist. Is it to make things easier when testing/developing? So you don't have to run EM step to check if these functions work in a pipeline?

Reply:
Yes, originally it was so we could run `linker.predict()` without any training. I haven't checked to see if this is now no longer a requirement.

Reply:
Ok, that feels like a wider consideration to this PR. Going to resolve this conversation, but have raised in #1104